code/Items/ItemStackWeapon.cs

```diff
@@ -1,15 +1,136 @@
-﻿using Sandbox;
+﻿using System.Collections.Generic;
+using Sandbox;
 using ssl.Items.Data;
 
 namespace ssl.Items
 {
-    public class ItemStackWeapon : ItemStack
+    public partial class ItemStackWeapon : ItemStack
     {
-        public BaseWeapon Weapon;
+        public virtual float PrimaryRate => 5.0f;
+        public override string ViewModelPath => "weapons/rust_pistol/v_rust_pistol.vmdl";
+        
+        public ItemStackWeapon()
+        {
+        }
+
+        public ItemStackWeapon(Item item, int amount = 1) : base(item, amount)
+        {
+        }
+        
+        public override void Spawn()
+        {
+            base.Spawn();
+
+            CollisionGroup = CollisionGroup.Weapon; // so players touch it as a trigger but not as a solid
+            SetInteractsAs(CollisionLayer.Debris); // so player movement doesn't walk into it
+        }
+
+        [Net, Predicted] private TimeSince TimeSincePrimaryAttack { get; set; }
+
+        public override void Simulate(Client player)
+        {
+            if (CanReload())
+            {
+                Reload();
+            }
+            
+            if (CanPrimaryAttack())
+            {
+                TimeSincePrimaryAttack = 0;
+                AttackPrimary();
+            }
+        }
+
+        protected virtual bool CanReload()
+        {
+            return Owner.IsValid() && Input.Down(InputButton.Reload);
+        }
+
+        protected virtual void Reload()
+        {
+        }
+
+        public virtual bool CanPrimaryAttack()
+        {
+
```

Comment:
How does it check that the weapon is selected ? Is it Owner.IsValid ?

Reply:
owner.valid check if the owner exists so yeah